core/lib/dal/src/eth_sender_dal.rs

```diff
@@ -163,18 +162,26 @@ impl EthSenderDal<'_, '_> {
         Ok(count.try_into().unwrap())
     }
 
-    pub async fn get_chain_id_of_last_eth_tx(&mut self) -> DalResult<Option<u64>> {
+    pub async fn get_chain_id_of_oldest_unfinalized_eth_tx(&mut self) -> DalResult<Option<u64>> {
         let res = sqlx::query!(
             r#"
             SELECT
                 chain_id
             FROM
                 eth_txs
-            ORDER BY id DESC
+            INNER JOIN l1_batches
+                ON
+                    eth_txs.id = l1_batches.eth_commit_tx_id
+                    OR eth_txs.id = l1_batches.eth_prove_tx_id
+                    OR eth_txs.id = l1_batches.eth_execute_tx_id
+            LEFT JOIN eth_txs_history ON eth_txs.id = eth_txs_history.eth_tx_id
+            WHERE
+                eth_txs_history.finality_status != 'finalized'
+            ORDER BY l1_batches.number ASC
```

Comment:
Why do you need l1 batch here? it's good enough to order based on eth_tx_id

Reply:
For example batch execute for batch X can have higher id then batch commit for batch Y, Y>X. Thats normal if it was performed later, it also happens on initial sync due to how batch_status_updater works. If there is a gateway migration between those, then we have a problem. Note that this SQL is called only once on node start.

Reply:
@Deniallugo is it consistency checker causing the restarts or what?

Reply:
Yeah, we have to check how it would work. But in the current code. it's ok